book/src/proposals/booting-from-snapshot.md

```diff
@@ -0,0 +1,222 @@
+# Booting From a Snapshot
+
+## Problem
+
+A validator booting from a snapshot should confirm that they are caught up to
+the correct cluster before voting. There are two major risks if they do not:
+
+### Wrong Cluster
+
+The validator may be fed a snapshot and gossip info for a different (though not
+necessarily malicious) cluster. If that cluster is at slot 100,000 while the
+cluster the validator intended to join is at 10,000, if the validator votes
+once on the wrong cluster, it is locked out of the correct cluster for at least
```

Comment:
How does the slot hash of the the vote on the slot 100,000 cluster match in the slot 10,000 cluster?

Reply:
My understanding is that the hash doesn't matter, any vote for a slot numbered 100,000 is slashable on the correct cluster.

Reply:
cc @aeyakovenko

Reply:
Seems like we are bringing up two different things. I may be wrong, but this is how I understand it: There is the tower lockout if you vote on a slot which is not picked as the main fork, then you cannot vote on the main fork for some duration of slots. This part is implemented but I don't see how this would get hit since the slot hashes would have to match at least in the current implementation.

Then there is slashing which is yet to be implemented and I guess we would have a rule about voting on different forks then you would be slashed and either lose your stake or your stake is locked for some period of time. It seems like we might need some kind of rule to distinguish between clusters because if there are two, then anyone could take one vote from one cluster and apply it to another one. Although it's not necessarily the case that the validator's key would have stake on the other one to be slashed, actually it seems unlikely unless the new cluster is a recent fork of the other one.

In any case, I think we should change the language to specify which of these two cases we are talking about, and maybe explain a bit how the slashing rule is applied or at least some link to a description of how the slashing rule works.

Reply:
Do we have a link to how the slashing rule works?

Reply:
@TristanDebrunner, still a proposal: https://github.com/solana-labs/solana/pull/6255.

Reply:
A vote on one cluster can be applied to another cluster as a slashing rule, but you don't care as long as you are safe on the main cluster. You want to ensure sure you don't vote on some bogus cluster that can be used to slash you on the main cluster.

Reply:
Clarified to specify slashing. No link yet, since we don't have an official one